app/models/account/syncer.rb

```diff
@@ -0,0 +1,53 @@
+class Account::Syncer
+    def initialize(account)
+      @account = account
+    end
+
+    def sync
+        sync_balances
+        purge_balances
+
+        Rails.logger.info("Synced account #{@account.name}")
+    end
+
+    private
+        # TODO: Support partial syncs via start_date
+        def sync_balances(start_date = nil)
+          sync_start_date = [ start_date, @account.effective_start_date ].compact.max
+
+          valuations = @account.valuations.where("date >= ?", sync_start_date).order(:date).select(:date, :value)
+          transactions = @account.transactions.where("date > ?", sync_start_date).order(:date).select(:date, :amount)
+          oldest_entry = [ valuations.first, transactions.first ].compact.min_by(&:date)
+
+          net_transaction_flows = transactions.sum(&:amount)
```

Comment:
since there is a currency attribute to transaction model, is it possible that we are adding amounts in mutiple currencies together?

Reply:
@sriram2520 yes, very good point—this PR assumes a single currency for an account.  

I'm going to add this point to the list of "Future improvements" (in PR description) as I think that would be a good isolated PR to open and nail down how we handle the conversion of "non-standard" transactions for an account.

Reply:
@zachgoll I can't think of a scenario where the transaction currency would be different from the account's currency. My suggestion would be to remove currency attribute from transaction model, as I dont see it being used anyway.

Reply:
@sriram2520 multi-currency accounts aren't super common, but they do exist and have enough adoption for us to support them (think Wise, Revolut)—https://www.nerdwallet.com/article/banking/multi-currency-account